eng/tools/GenerateFiles/Directory.Build.props.in

```diff
@@ -8,4 +8,11 @@
     <!-- Temporarily hardcoded to true -->
     <SuppressGenerateILCompilerExplicitPackageReferenceWarning>true</SuppressGenerateILCompilerExplicitPackageReferenceWarning>
   </PropertyGroup>
+
+  <!-- Ignore warning about RID resolution rules changing in .NET 8 Preview 6 -->
+  <!-- Needed until https://github.com/ericsink/SQLitePCL.raw/issues/543 is fixed -->
+  <!-- See https://github.com/dotnet/aspnetcore/pull/48908#issuecomment-1601894643 -->
+  <ItemGroup>
```

Comment:
How does this work for templates?

Reply:
I thought it might work for the templates because of these lines:

https://github.com/dotnet/aspnetcore/blob/9f4b8b4c7594f50a2ad5746c4c2f7c8c2d87f750/src/ProjectTemplates/TestInfrastructure/PrepareForTest.targets#L85-L88

https://github.com/dotnet/aspnetcore/blob/9f4b8b4c7594f50a2ad5746c4c2f7c8c2d87f750/src/ProjectTemplates/TestInfrastructure/Directory.Build.props.in#L8

But I was just trying thing, and I'm a bit surprised it worked too. @wtgodbe Is there a better way to do this? I saw that @mitchdenny updated this a couple months ago as part of #48014. Do we still need to hardcode SuppressGenerateILCompilerExplicitPackageReferenceWarning?

Reply:
Yeah, this is the right way to do it, at least for in-repo stuff. Template tests will get the workaround, but template projects may run into issues in real world scenarios (since the workaround won't be injected into them). I don't know enough about this specific issue to know if it's something that would actually manifest.

@eerhardt do we still need the `SuppressGenerateILCompilerExplicitPackageReferenceWarning` workaround?

Reply:
> Yeah, this is the right way to do it, at least for in-repo stuff. Template tests will get the workaround, but template projects may run into issues in real world scenarios (since the workaround won't be injected into them). I don't know enough about this specific issue to know if it's something that would actually manifest.

Ah, that makes sense. Hopefully, this is resolved for preview7 and we don't have to issue a known issues warning. 🤞🏽

Reply:
> @eerhardt do we still need the SuppressGenerateILCompilerExplicitPackageReferenceWarning workaround?

I believe so. I don't know of any changes here that would cause us to no longer need the workaround. We are running our tests against a specific version of the runtime, and not the runtime that comes with the SDK. See https://github.com/dotnet/runtime/issues/84372.

Reply:
> Ah, that makes sense. Hopefully, this is resolved for preview7 and we don't have to issue a known issues warning. 🤞🏽

I doubt this will be fixed by preview7. Even if https://github.com/ericsink/SQLitePCL.raw/issues/543 is resolved by then and uploaded to nuget, EF still needs to update its dependencies. Anyone using Microsoft.Data.Sqlite will likely see warnings until rc1 at the earliest unless they manually upgrade, so we should probably get ahead of the issues. @dotnet/efteam

Reply:
I wonder if it would be reasonable to add a mechanism to tell the SDK that it should not warn on some particular package or particular versions of packages - for example, the consumer has checked that they are not affected by the non-portable RID in that package, as would be the case here. @dsplaisted @vitek-karas @agocke - thoughts?

It would go kind of counter to making a hard push for packages to fix their RID usage, but it would provide an option between suppressing the warning entirely and switching to the old behaviour - neither of which we really want people to do.

Reply:
Is there a way to suppress the warning by its warning code - I don't know if `NoWarn` works on SDK warnings. If that works, then I'd probably not add any new support and ask the affected people to suppress the warning code instead.

Basically we want packages to fix this problem, and there are not that many packages which are affected by this and consequently not that many teams/customers will see the warning either. I'd like to avoid adding production code into the SDK unless there's a strong reason for it - and this doesn't feel that strong - assuming the warning can be suppressed already.

Reply:
`<NoWarn>NETSDK1206</NoWarn>` works.

Reply:
> Basically we want packages to fix this problem, and there are not that many packages which are affected by this and consequently not that many teams/customers will see the warning either.

That's fair. The main concern is that the affected package is brought in as a dependency under certain template configurations to there is a chance that users might run into the issue with this particular package at a higher rate than other packages.